test/end-to-end/core.sh

```diff
@@ -84,193 +85,194 @@ wait_for_registry
+os::cmd::expect_success_and_text "cat '${LOG_DIR}/cli-with-token.log'" 'Using in-cluster configuration'
+os::cmd::expect_success_and_text "cat '${LOG_DIR}/cli-with-token.log'" 'In project test'
+os::cmd::expect_success 'oc delete pod cli-with-token'
+
+os::cmd::expect_success "oc run cli-with-token-2 --attach --env=POD_NAMESPACE=test --image=openshift/origin:${TAG} --restart=Never -- cli whoami --loglevel=4 > '${LOG_DIR}/cli-with-token2.log' 2>&1"
+os::cmd::expect_success_and_text "cat '${LOG_DIR}/cli-with-token2.log'" 'system:serviceaccount:test:default'
+os::cmd::expect_success 'oc delete pod cli-with-token-2'
+
+os::cmd::expect_success "oc run kubectl-with-token --attach --env=POD_NAMESPACE=test --image=openshift/origin:${TAG} --restart=Never --command -- kubectl get pods --loglevel=4 > '${LOG_DIR}/kubectl-with-token.log' 2>&1"
+os::cmd::expect_success_and_text "cat '${LOG_DIR}/kubectl-with-token.log'" 'Using in-cluster configuration'
+os::cmd::expect_success_and_text "cat '${LOG_DIR}/kubectl-with-token.log'" 'kubectl-with-token'
 
 echo "[INFO] Streaming the logs from a deployment twice..."
-oc create -f test/fixtures/failing-dc.yaml
-tryuntil oc get rc/failing-dc-1
-oc logs -f dc/failing-dc
-wait_for_command "oc get rc/failing-dc-1 --template={{.metadata.annotations}} | grep openshift.io/deployment.phase:Failed" $((60*TIME_SEC))
-oc logs dc/failing-dc | grep 'test pre hook executed'
-oc deploy failing-dc --latest
-oc logs --version=1 dc/failing-dc
+os::cmd::expect_success 'oc create -f test/fixtures/failing-dc.yaml'
+os::cmd::try_until_success 'oc get rc/failing-dc-1'
+os::cmd::expect_success 'oc logs -f dc/failing-dc'
+os::cmd::try_until_text 'oc get rc/failing-dc-1 --template={{.metadata.annotations}}' 'openshift.io/deployment.phase:Failed'
+os::cmd::expect_success_and_text 'oc logs dc/failing-dc' 'test pre hook executed'
+os::cmd::expect_success 'oc deploy failing-dc --latest'
+os::cmd::expect_success 'oc logs --version=1 dc/failing-dc'
 
 echo "[INFO] Applying STI application config"
-oc create -f "${STI_CONFIG_FILE}"
+os::cmd::expect_success "oc create -f ${STI_CONFIG_FILE}"
 
 # Wait for build which should have triggered automatically
 echo "[INFO] Starting build from ${STI_CONFIG_FILE} and streaming its logs..."
-#oc start-build -n test ruby-sample-build --follow
-os::build:wait_for_start "test"
+os::cmd::try_until_text 'oc get -n test builds' 'Running'
+ 
 # Ensure that the build pod doesn't allow exec
-[ "$(oc rsh ${BUILD_ID}-build 2>&1 | grep 'forbidden')" ]
-os::build:wait_for_end "test"
+BUILD_ID=$(oc get -n test builds  --output-version=v1 --template="{{with index .items 0}}{{.metadata.name}}{{end}}")
+os::cmd::expect_failure_and_text "oc rsh ${BUILD_ID}-build" 'forbidden'
+os::cmd::try_until_text 'oc get -n test builds' 'Complete' $((10*TIME_MIN))
+os::cmd::expect_success "oc build-logs -n test ${BUILD_ID} > '${LOG_DIR}/testbuild.log'" # TODO(skuznets): previously we were agnostic about this failing, why?
```

Comment:
Re-commenting since line was changed, comment on previous diff lost.

@deads2k previously we were agnostic about the `oc build-logs` failing - why? In what case do we want to continue the test when getting logs fails?